### ISSUE TITLE
`SelectionPrompt` does not search on results of `UseConverter`

### ISSUE THREAD

**Issue by snus-kin** [spectreconsole/spectre.console#1626]
**Information**
 - OS: MacOS (also tested on Windows)
 - Version: 0.49.1
 - Terminal: IntelliJ Rider Built-in Terminal (using Bash)

**Describe the bug**
Selection prompts which use `UseConverter` do not jump to the search (they do not use the mapped text as the search index).

**To Reproduce**
- Create a selection prompt
- Use .UseConverter()
- Search
- Observe that the result you have searched for is not jumped to

You can also search for the underlying object and observe that it is jumped to.

Example unit test that currently fails (where `MyClass` is just a class with a name parameter):
```c#
    [Fact]
    public void Should_Search_In_Remapped_Result()
    {

        // Given
        var console = new TestConsole();
        console.Profile.Capabilities.Interactive = true;
        console.EmitAnsiSequences();
        console.Input.PushText("2");
        console.Input.PushKey(ConsoleKey.Enter);
        var myObject = new List<MyClass>{ new() { name = "Item 1" }, new() {name = "Item 2"} };

        // When
        var prompt = new SelectionPrompt<MyClass>()
            .Title("Select one")
            .EnableSearch()
            .UseConverter(o => o.name)
            .AddChoices(myObject);
        var selection = prompt.Show(console);

        // Then
        selection.name.ShouldBe("Item 2");
    }
```

**Expected behavior**
SelectionPrompt should search based on the results of `UseConverter` and not on the underlying object remapped by it. (see test)

**Screenshots**

**Additional context**

---
Please upvote :+1: this issue if you are interested in it.

**Comment by snus-kin** [comment]
Some testing, this test works: 

```c#
    [Fact]
    public void Should_Search_In_Remapped_Result()
    {

        // Given
        var console = new TestConsole();
        console.Profile.Capabilities.Interactive = true;
        console.EmitAnsiSequences();
        console.Input.PushText("2");
        console.Input.PushKey(ConsoleKey.Enter);

        var myObject = new List<Tuple<string, string>>
        {
            new ( "Value", "Item 1" ),
            new ( "Value", "Item 2" )
        };

        // When
        var prompt = new SelectionPrompt<Tuple<string, string>>()
            .Title("Select one")
            .EnableSearch()
            .UseConverter(o => o.Item2)
            .AddChoices(myObject);
        var selection = prompt.Show(console);

        // Then
        selection.Item2.ShouldBe("Item 2");
    }
```

**Comment by snus-kin** [comment]
Potential fix here: https://github.com/snus-kin/spectre.console/commit/b1d8d6d6edf8332798113701dfb0bf9356a91687

Not sure if there's a better way of doing this, tried to refactor so the function is only called once per item. Will wait for a reply on this issue before creating a PR or anything.

**Comment by patriksvensson** [comment]
Thanks for attempting to fix this, but we don't want to change the API. The UseConverter is used to get a representation, so we do not want to add an explicit representation when adding the item (optional or not) since this is what the converter is for. I will look into what is required to fix the search.

**Comment by patriksvensson** [comment]
@snus-kin I've submitted a PR that should fix this. Will merge shortly

**Comment by patriksvensson** [comment]
@snus-kin This is fixed and is now available in `0.49.2-preview.0.26`

**Comment by snus-kin** [comment]
Thanks pal, appreciate it
